src/DataGrid.tsx

```diff
@@ -904,6 +904,7 @@ export function DataGrid<R, SR = unknown, K extends Key = Key>(props: DataGridPr
     const { idx, row } = selectedPosition;
     const column = columns[idx];
     const colSpan = getColSpan(column, lastFrozenColumnIndex, { type: 'ROW', row });
+    const discardOnRowChange = column.editorOptions?.discardOnRowChange !== false;
```

Comment:
Maybe this is clearer, no?
```suggestion
    const discardOnRowChange = column.editorOptions?.discardOnRowChange ?? true;
```